src/core/public/chrome/ui/loading_indicator.tsx

```diff
@@ -39,16 +39,26 @@ export class LoadingIndicator extends React.Component<LoadingIndicatorProps, { v
     visible: false,
   };
 
+  private timer: any;
+  private increment = 1;
+
   componentDidMount() {
     this.loadingCountSubscription = this.props.loadingCount$.subscribe((count) => {
-      this.setState({
-        visible: count > 0,
-      });
+      if (this.increment > 1) {
+        clearTimeout(this.timer);
+      }
+      this.increment += this.increment;
```

Comment:
As `increment` is initialized to `1`, I'm ensure to understand what exactly this block does. Both the `if (this.increment > 1)` (that would be always `true`) and `this.increment += this.increment;` seems wrong to me?

Reply:
It makes sure that the 250 pause only starts the first time the cycle begins. You can see that it does print initially and only increases when the subscription does.

The subscription code isn't the most approachable, but this method seemed to do what was needed: only show a loading indicator if a new "loading event" existed for more than 250ms.

![image](https://user-images.githubusercontent.com/324519/94695875-4d287800-02eb-11eb-8fe0-eea5b764cbb3.png)